### PR TITLE
[OTA] Add On/Off Light Device Type to ota-requestor-app

### PR DESCRIPTION
**Pull request by carol-apple**
#### Problem
The original purpose of ota-requestor-app is to create a sample app that contains the OTA Requestor Server cluster to be used for verifying OTA functionality. However, in reality, an accessory with only the OTA Requestor Server cluster and no other clusters does not make sense.

#### Change overview
Add an On/Off Light Device Type to make ota-requestor-app a more realistic accessory

#### Testing
Tree compiles and app can be commissioned

### CODE REVIEW

File: examples/ota-requestor-app/ota-requestor-common/ota-requestor-app.matter [hunk 1]
```diff
@@ -6,6 +6,168 @@ struct LabelStruct {
     char_string<16> value = 1;
 }
 
+server cluster Identify = 3 {
```
Comment (bzbarsky-apple) [inline examples/ota-requestor-app/ota-requestor-common/ota-requestor-app.matter:9]: This seems fine as far as it goes, but don't we need to actually compile in the relevant cluster impls?

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/ota-requestor-app/ota-requestor-common/ota-requestor-app.matter (hunk 1 above)
Comment (carol-apple) [reply examples/ota-requestor-app/ota-requestor-common/ota-requestor-app.matter:9]: Yeah, these are automatically compiled in for the gn-based apps. I added it for esp32.

File: examples/ota-requestor-app/esp32/main/CMakeLists.txt [hunk 2]
```diff
@@ -38,7 +38,10 @@ set(SRC_DIRS_LIST
                       "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/descriptor"
                       "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/diagnostic-logs-server"
                       "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/ethernet-network-diagnostics-server"
+                      "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/groups-server"
```
Comment (harsha-rajendran) [inline examples/ota-requestor-app/esp32/main/CMakeLists.txt:41]: Just checking- are there any similar build-related changes that might need to be made for other platforms?

**Review by harsha-rajendran — APPROVED** [review]

File: examples/ota-requestor-app/esp32/main/CMakeLists.txt (hunk 2 above)
Comment (carol-apple) [reply examples/ota-requestor-app/esp32/main/CMakeLists.txt:41]: CI caught this one. I think it would have failed for other platforms if it was needed.

**Pull request merged**